tried to change the onChange in team search box

ItayNaumann opened the pull request.

ItayNaumann commented:
got a bit long there, will add the firebase changes in another commit.

itamarsch commented:
This PR looks Huge but it's mostly ui changes so git says it's a alot

itamarsch commented on lib/views/mobile/firebase_submit_button.dart:
```diff
@@ -124,7 +122,7 @@ class _FireBaseSubmitButtonState extends State<FireBaseSubmitButton> {
       } else if (event.state == TaskState.success) {
         final Map<String, dynamic> vars =
-            Map<String, dynamic>.from(widget.vars.toHasuraVars());
+            Map<String, dynamic>.from(widget.toHasuraVars());
```
You don't need this call to `from`

itamarsch commented on lib/views/pc/team_info/team_info_screen.dart:
```diff
@@ -8,63 +8,62 @@ import "package:scouting_frontend/views/common/dashboard_scaffold.dart";
+            final Widget searchBox,
+            final _,
+          ) =>
+              Column(
+            children: <Widget>[
+              Row(
+                children: <Widget>[
+                  Expanded(flex: 1, child: searchBox),
+                  SizedBox(width: defaultPadding),
+                  Expanded(
+                    flex: 2,
+                    child: Container(),
+                  ),
+                ],
+              ),
+              SizedBox(height: defaultPadding),
+              Expanded(
+                flex: 10,
+                child: team.mapNullable(
+                      TeamInfoData.new,
+                    ) ??
+                    NoTeamSelected(),
+              ),
+            ],
+          ),
```
Make this a function and have the function receive a body (in this case a `TeamInfo` or `NoTeamSelected`) and a searchBox
and then pass a call to the function with the correct parameters for buildWithTeam and buildWithoutTeam

giladwo commented:
![image](https://user-images.githubusercontent.com/25708271/216642581-7fe83ece-bde6-4fab-891b-b632116d8131.png)
![image](https://user-images.githubusercontent.com/25708271/216643161-0704a038-cba1-4e4f-877f-50da0a4c93c0.png)

There are way too many changes here, and they probably don't belong on the same PR.
Unless you can convince me they do, please split them into smaller PRs (shouldn't take long, provided the changes were 
carefully committed separately).

giladwo commented:
We don't want a mega PR that handles all open issues. We want small PRs, even multiple ones per issue (when it makes sense, i.e. a methodological change that should be applied to different parts).

ItayNaumann commented:
![image](https://user-images.githubusercontent.com/93080393/216808562-a327317d-c9a9-4132-8bdf-202049e72a18.png)
![image](https://user-images.githubusercontent.com/93080393/216808771-4c6af7c6-a5ef-431b-8b74-2c83158faaaa.png)
we are removing here not needed default values to send hasura,setting them as the default for the class and removing the to hasura vars function and moving it to the only time we use it and removing the team from the class. probably not needed in this pr should be on another pr.

ItayNaumann commented:
![image](https://user-images.githubusercontent.com/93080393/216808912-d5c86184-ef60-495c-a503-72b9620a50d4.png)
switched to the new way we are using teamSelectionFuture

ItayNaumann commented:
![image](https://user-images.githubusercontent.com/93080393/216808944-bd207918-ab7a-4f6c-b768-5d5f92c54bab.png)
![image](https://user-images.githubusercontent.com/93080393/216808959-d2bfc84f-0bbc-4a6e-b29e-db5c27bdfc95.png)
![image](https://user-images.githubusercontent.com/93080393/216809419-6df51a72-00b9-4c77-8f5d-d9c9f4eac555.png)
this is the main change of the pr, changing teamSearchBox, in here we are getting from where its called a couple of functions.
setTeam is to be able to have a onSelected cause we are setting up the widget here so the place it is called to be cant access it without it.
on build we have 2 functions; build with team and build without team. we are calling build with team when we have team and when its null we are calling build without team.
also using the setTeam function instead of using setState directly.

ItayNaumann commented:
![image](https://user-images.githubusercontent.com/93080393/216809696-46883a9b-9fa5-47c9-aba2-4ac52c74ba37.png)
in here we probably went too far for this pr; removed the HasuraVars file and changed the submit button to get a toHasuraVars function

ItayNaumann commented:
![image](https://user-images.githubusercontent.com/93080393/216809863-8f96f770-ad8a-4e16-a26e-9dd36600497c.png)
removed the toHasura vars in pit vars and removed the team from pit vars

ItayNaumann commented:
![image](https://user-images.githubusercontent.com/93080393/216812149-c9129248-2c39-41e5-8952-29ce39d57504.png)
in here (fault view) its just changed to the new way we use teamSelectionFuture and we removed team from it

ItayNaumann commented:
in input view the git diff is no better than the normal diff, but we just removed lightTeam from there, changed to the new way it shows only the match and team search boxes, but then it builds the whole screen after he gets both of these variables,also defined the parsing "toHasuraVars" in the file instead of in the vars of  this file

ItayNaumann commented:
![image](https://user-images.githubusercontent.com/93080393/216812950-c6c6c47d-e96c-4a04-8248-f98a8a283776.png)
changed to new teamSelectionFuture, changed from a list of controllers to each controller by himself and moved the parse to hasura vars to here

ItayNaumann commented:
in specific view its pretty much the same as input view; removed the list of controllers, changed to new teamSelectionFuture version and moved the definition of the toHasuraVars to here and changed that the screen wouldnt be shown until the user inputed a match and a team

ItayNaumann commented:
side nav bar just changed to new team selecitonFuture
![image](https://user-images.githubusercontent.com/93080393/216813280-3a296f00-2493-4373-8e62-2375c7c4a48c.png)

ItayNaumann commented:
![image](https://user-images.githubusercontent.com/93080393/216813296-63fa6947-1dea-49ab-a7a6-22374e0ceb1c.png)
in specific vars simply removed the toHasuraVars function and removed lightTeam from being defined there

ItayNaumann commented:
![image](https://user-images.githubusercontent.com/93080393/216813325-50c679c2-6e6b-4668-8785-0b829dc4a884.png)
in the submit button just changed it from getting a map of string to dynamic to a function that returns that map

ItayNaumann commented:
![image](https://user-images.githubusercontent.com/93080393/216813385-973c6e22-9030-46ea-9a23-611e3053be3d.png)
in teamSelectionMatches just changed to include new features from teamSelectionFuture

ItayNaumann commented:
![image](https://user-images.githubusercontent.com/93080393/216813467-83700aee-c222-4977-bdf5-9a9baa78fa86.png)
in compare screen we remove all of the controllers from this scope and add before the return of the search box the teams that arent in the list full of teams

ItayNaumann commented:
![image](https://user-images.githubusercontent.com/93080393/216813955-7a7856af-dd87-4efd-bc89-649b371cf2ce.png)
![image](https://user-images.githubusercontent.com/93080393/216813968-1ae196ee-e5e1-4bee-938f-0d2fa12b9b7d.png)
![image](https://user-images.githubusercontent.com/93080393/216813976-e0d999d1-b205-411a-8ca2-73e852a745c2.png)
making an empty array of void functions that reset the search box and defining each one of them to a different search box and then after we send the info to the db we reset all of them at once

ItayNaumann commented:
![image](https://user-images.githubusercontent.com/93080393/216814115-6c997d6a-baf1-46cc-89a3-ee7cf62bd93f.png)
removing the sending to db here...

ItayNaumann commented:
![image](https://user-images.githubusercontent.com/93080393/216814163-10f7ed25-a2ee-4e78-870f-727c606ecb94.png)
changing to new way of teamSelectionFuture

ItayNaumann commented:
I think i can make this pr 2 pr, one for removing the hasura things and one for all the others

amitcha06 commented:
> ![image](https://user-images.githubusercontent.com/93080393/216809696-46883a9b-9fa5-47c9-aba2-4ac52c74ba37.png) in here we probably went too far for this pr; removed the HasuraVars file and changed the submit button to get a toHasuraVars function

Yeah These should NOT be on the same PR

amitcha06 commented:
> in input view the git diff is no better than the normal diff, but we just removed lightTeam from there, changed to the new way it shows only the match and team search boxes, but then it builds the whole screen after he gets both of these variables,also defined the parsing "toHasuraVars" in the file instead of in the vars of this file

Note that #356 is not merged yet, so there will be merge conflicts

amitcha06 commented:
> in specific view its pretty much the same as input view; removed the list of controllers, changed to new teamSelectionFuture version and moved the definition of the toHasuraVars to here and changed that the screen wouldnt be shown until the user inputed a match and a team

i would argue that anything unrelated to TeamSelection should not be in this PR (the controllers, for example)

amitcha06 commented:
> ![image](https://user-images.githubusercontent.com/93080393/216813296-63fa6947-1dea-49ab-a7a6-22374e0ceb1c.png) in specific vars simply removed the toHasuraVars function and removed lightTeam from being defined there

Same as stated above, the HasuraVars shouldnt be in this PR

amitcha06 commented:
> ![image](https://user-images.githubusercontent.com/93080393/216813325-50c679c2-6e6b-4668-8785-0b829dc4a884.png) in the submit button just changed it from getting a map of string to dynamic to a function that returns that map

^

amitcha06 commented:
> ![image](https://user-images.githubusercontent.com/93080393/216813955-7a7856af-dd87-4efd-bc89-649b371cf2ce.png) ![image](https://user-images.githubusercontent.com/93080393/216813968-1ae196ee-e5e1-4bee-938f-0d2fa12b9b7d.png) ![image](https://user-images.githubusercontent.com/93080393/216813976-e0d999d1-b205-411a-8ca2-73e852a745c2.png) ![image](https://user-images.githubusercontent.com/93080393/216814092-5dde3342-ea13-4cc4-abf6-ba7499a28121.png) making an empty array of void functions that reset the search box and defining each one of them to a different search box and then after we send the info to the db we reset all of them at once and defining the sending to db here..

These should also be split

amitcha06 commented:
> ![image](https://user-images.githubusercontent.com/93080393/216814115-6c997d6a-baf1-46cc-89a3-ee7cf62bd93f.png) removing the sending to db here...

^

amitcha06 commented:
> I think i can make this pr 2 pr, one for removing the hasura things and one for all the others

They can probably even more. 
one for the TeamSelection
one for HasuraVars
one for the changes in matches_screen,
etc...
